components/brave_federated/operational_patterns_util_unittest.cc

```diff
@@ -0,0 +1,180 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/operational_patterns_util.h"
+
+#include <map>
+#include <string>
+
+#include "base/test/scoped_feature_list.h"
+#include "base/time/time.h"
+#include "brave/components/brave_federated/features.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+// npm run test -- brave_unit_tests
+// -filter=BraveFederatedOperationalPatternsUtilTest*
+
+namespace brave_federated {
+
+class BraveFederatedOperationalPatternsUtilTest : public testing::Test {
+ public:
+  BraveFederatedOperationalPatternsUtilTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {}
+
+ protected:
+  content::BrowserTaskEnvironment task_environment_;
+};
+
+TEST_F(BraveFederatedOperationalPatternsUtilTest,
+       GetCurrentCollectionSlotAtTheBeginningOfTheMonth) {
+  // Arrange
+  const base::Time::Exploded exploded_time = {
+      /* year */ 2022,
+      /* month */ 1,
+      /* day_of_week */ 0,
```

Comment:
January 1st 2022 was a Saturday, should we not set to the correct value?

Reply:
I couldn't bother to look them up, but added correct dow

Reply:
It may not be important :-)